contracts/protocol/ProtocolImplementation.sol

```diff
@@ -0,0 +1,99 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.26;
+
+import "./ScoutTokenERC20.sol";
+import "./libs/MemoryUtils.sol";
+import "./libs/ProtocolAccessControl.sol";
+import "@openzeppelin/contracts/utils/Context.sol";
+import "@openzeppelin/contracts/utils/cryptography/MerkleProof.sol";
+import "@openzeppelin/contracts/utils/StorageSlot.sol";
+
+contract ScoutProtocolImplementation is Context, ProtocolAccessControl {
+    using MemoryUtils for bytes32;
+
+
+    modifier onlyAdminOrClaimManager() {
+      require(_hasRole(MemoryUtils.CLAIM_MANAGER_SLOT) || _isAdmin() , "Proxy: caller is not the claim manager");
+        _;
+    }
+
+    // Allow the sender to claim their balance as ERC20 tokens
+    function claim(
+        string memory week,
+        uint256 amount,
+        bytes32[] calldata proofs
+    ) public onlyWhenNotPaused returns (bool)  {
+        // Check if the user has already claimed for the given week
+        require(!hasClaimed(week, msg.sender), "You have already claimed for this week.");
+
+        // Get the Merkle root for the given week
+        bytes32 merkleRoot = getMerkleRoot(week);
+
+        // Ensure the Merkle root is set
+        require(merkleRoot != bytes32(0), "Merkle root for this week is not set.");
+
+        // Construct the leaf node from the user's address and the amount
+        bytes32 leaf = keccak256(abi.encodePacked(msg.sender, amount));
+
+        // Verify the Merkle proof
+        require(MerkleProof.verify(proofs, merkleRoot, leaf), "Invalid Merkle proof.");
+
+        // Mark the user as having claimed for this week
+        setClaimed(week, msg.sender);
+
+        // Ensure the contract has enough tokens to fulfill the claim
+        uint256 contractBalance = _getToken().balanceOf(address(this));
+        require(contractBalance >= amount, "Insufficient balance in contract.");
+
+        ScoutTokenERC20 token = _getToken();
+
+        // Transfer tokens to the user
+        token.transfer(msg.sender, amount * (10 ** token.decimals()));
```

Comment:
For easier reasoning, we pass the amount as integers across systems

We only need to wei-ify amounts in the contract